src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -563,6 +561,46 @@ public void run() {
         return Collections.emptyList();
     }
 
+    /**
+     * Returns the full bundle name.
+     *
+     * @return the full bundle name.
+     */
+    @NonNull
+    public static String getBundleFileName() {
+        StringBuilder filename = new StringBuilder();
+        filename.append(getBundlePrefix());
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd_HH.mm.ss");
+        dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
+        filename.append("_").append(dateFormat.format(new Date()));
+
+        filename.append(".zip");
+        return filename.toString();
+    }
+
+    /**
+     * Returns the prefix of the bundle name.
+     *
+     * @return the prefix of the bundle name.
+     */
+    private static String getBundlePrefix() {
+        String filename = "support"; // default bundle filename
+        final SupportPlugin instance = getInstance();
+        if (instance != null) {
+            SupportProvider supportProvider = instance.getSupportProvider();
+            if (supportProvider != null) {
+                // let the provider name it
+                filename = supportProvider.getName();
```

Comment:
Still leaves me wondering about why we have both systems. In the case of a vendor Acme implementing both extension points in a Blaster product, we will get something like

    acme-support_acme-blaster_2016-12-07_16.12.47.zip

which seems redundant.